vnext/Shared/HermesRuntimeHolder.cpp

```diff
@@ -29,6 +31,55 @@ namespace {
 std::unique_ptr<facebook::hermes::HermesRuntime> makeHermesRuntimeSystraced(
     const ::hermes::vm::RuntimeConfig &runtimeConfig) {
   SystraceSection s("HermesExecutorFactory::makeHermesRuntimeSystraced");
+
+  auto res = LoadLibrary(L"hermes.dll");
+  if (!res) {
+    DWORD lastError = GetLastError();
+    std::stringstream ostr;
+    ostr << "Failure loading hermes.dll:" << std::hex << lastError;
+    facebook::react::tracing::error(ostr.str().c_str());
+  } else {
+    facebook::react::tracing::error("hermes.dll loaded successfully");
+  }
+
+  auto res2 = LoadLibraryEx(L"hermes.dll", NULL, DONT_RESOLVE_DLL_REFERENCES);
+  if (!res2) {
+    DWORD lastError = GetLastError();
+    std::stringstream ostr;
+    ostr << "Failure loading hermes.dll even without resolving references:" << std::hex << lastError;
+    facebook::react::tracing::error(ostr.str().c_str());
+  } else {
+    facebook::react::tracing::error("hermes.dll loaded successfully without resolving references");
+  }
+
+  auto res3 = LoadLibrary(L"icu.dll");
+  if (!res3) {
+    DWORD lastError = GetLastError();
+    std::stringstream ostr;
+    ostr << "Failure loading icu.dll:" << std::hex << lastError;
+    facebook::react::tracing::error(ostr.str().c_str());
+  } else {
+    facebook::react::tracing::error("icu.dll loaded successfully");
+  }
+
+  auto res4 = LoadLibrary(L"c:\\windows\\system32\\icu.dll");
```

Comment:
>"c: [](http://example.com/codeflow?start=27&length=3)

What if Windows is installed somewhere else? A better way would be to use an environment variable %WINDIR% or %SYSTEMROOT%.

Reply:
I also think that this call us reduntant because the LoadLibrary("icu.dll") above checks the system32 in case if it is not found locally.
I would call it also dangerous to load the icu.dll from local and then from the system dir. We may end up loading two DLLs with different versions and some unpredicted behavior.

Reply:
I don't think we should search anything except for system32, so we should use `LoadLibraryEx` with `LOAD_LIBRARY_SEARCH_SYSTEM32`

Reply:
I think we can just always rely on ICUUC.dll, since it forwards to ICU.dll on recent versions, and works downlevel too, so no need to try ICU separately

Reply:
@vmoroz @asklar Sorry, all these code was added to diagnose the dll load failure on the CI machines, but not our dev machines .. It turned out that the CI machines are based on 2019 datacenter SKU which was based on Windows RS5 (1809) bits  .. hence don't contain "icu.dll", but contains "icuuc.dll" and "icuin.dll" .. 

I've created a new hermes package with some adjustments in the build scripts to not try to load "icu.dll". And, I'm removing all these diagnostics in the latest PR ..